source/BufferParser.cpp

```diff
@@ -116,13 +117,24 @@ void BufferParser::addCodecDataToSegment(std::unique_ptr<firebolt::rialto::IMedi
             GstMappedBuffer mappedBuf(buf, GST_MAP_READ);
             if (mappedBuf)
             {
-                segment->setCodecData(
-                    std::make_shared<std::vector<std::uint8_t>>(mappedBuf.data(), mappedBuf.data() + mappedBuf.size()));
+                auto codecData = std::make_shared<firebolt::rialto::CodecData>();
+                codecData->data = std::vector<std::uint8_t>(mappedBuf.data(), mappedBuf.data() + mappedBuf.size());
+                codecData->type = firebolt::rialto::CodecDataType::BUFFER;
+                segment->setCodecData(codecData);
             }
             else
             {
                 GST_ERROR("Failed to read codec_data");
             }
+            return;
+        }
+        const gchar *str = g_value_get_string(codec_data);
```

Comment:
I think that it can be in else clause. There's no point of checking if it's string if it's already a buffer.

Reply:
That's why we have return in line 129 :) But I will change it to else, if You prefer to have it.

Reply:
Ah, it's fine. I didn't notice the return in line 129 :P (Coming back from holidays is not an easy task :()